paimon-core/src/main/java/org/apache/paimon/utils/JsonSerdeUtil.java

```diff
@@ -219,6 +221,43 @@ public static <T> JsonNode toTree(T value) {
         return OBJECT_MAPPER_INSTANCE.valueToTree(value);
     }
 
+    /**
+     * Adds primary key fields to a JSON string.
+     *
+     * @param value The original JSON string.
+     * @param pkNames A list of primary key names to be added to the JSON string.
+     * @param pkNamesKey The key under which the primary key names will be added.
+     * @return The JSON string with the added primary key names. If the JSON string is not a valid
+     *     JSON object, or if the list of primary key names is empty or null, the original JSON
+     *     string will be returned.
+     * @throws RuntimeException If an error occurs while parsing the JSON string or adding the
+     *     primary key names.
+     */
+    public static String addPrimaryKeysToJson(
+            String value, List<String> pkNames, String pkNamesKey) {
```

Comment:
We can make this method more common. How about `String putArrayToJsonString(String origin, String key, List<String> values)`

Reply:
done.